src/Engine/EmitMSIL/Replication.cs

```diff
@@ -38,7 +46,12 @@ internal static List<CLRStackValue> MarshalFunctionArguments(IList args, MSILRun
             return stackValues;
         }
 
-        internal static List<object> UnmarshalFunctionArguments(List<CLRFunctionEndPoint.ParamInfo> formalParams, List<CLRStackValue> args, MSILRuntimeCore runtimeCore)
+        public static IList<object> UnMarshalFunctionArguments2(IEnumerable<CLRFunctionEndPoint> feps, IEnumerable args, MSILRuntimeCore runtimeCore)
+        {
+            return UnmarshalFunctionArguments(feps.FirstOrDefault().FormalParams, (IList)args, runtimeCore).ToArray();
```

Comment:
are we confident that the first FEP is good enough for these purposes ?
Should we add a TODO to figure out how to handle multiple FEPs ?

Reply:
I am not confident - I will add a TODO and look at this as a potential cause of test failures perhaps.

Reply:
As part of my task, I'm working on this to determine a single fep at compile time. If I'm not able to narrow it down to a single fep, I'm passing the list of feps for it to be determined at runtime.